### PR TITLE
Iteration 4 merged with functional errors in Elevator

### PR DESCRIPTION
**Pull request by EvanB213**

### CODE REVIEW

File: src/Elevator.java [hunk 1]
```diff
@@ -117,6 +119,7 @@ public void run() {
                 pause();
             }else {
                 int floorNum = requested.removeFirst();
+                System.out.println("Elevator MOVING TO " + floorNum);
```
Comment (boriszugic) [inline src/Elevator.java:122]: Remove

File: src/Elevator.java [hunk 2]
```diff
@@ -160,6 +154,8 @@ public synchronized void parseRequest(DatagramPacket packet){
                     }
             }
         }
+        requested.add(floorNum);
+        System.out.println("System test" + requested);
```
Comment (boriszugic) [inline src/Elevator.java:158]: Remove

File: src/Elevator.java [hunk 3]
```diff
@@ -130,23 +133,14 @@ public void run() {
...
-        /**Request sorting algorithm --incomplete **/
-//        requested.sort(Comparator.comparingInt(floor -> {
-//            int distance = Math.abs(floor - currentFloor);
-//            boolean isDirectionUp = floor > currentFloor;
-//            return isDirectionUp ? distance : -distance;
-//        }));
-
+        lamps.get(floorNum - 1).turnOn();
         if((int)packet.getData()[2] != 0){
             switch(packet.getData()[2]){
```
Comment (boriszugic) [inline src/Elevator.java:138]: Change to if statement

File: src/Elevator.java [hunk 4]
```diff
@@ -172,8 +168,9 @@ public void move(int floorNum){
         logger.debug("Closing door of elevator:" + id);
         door.close();
         logger.debug("Moving to floor " + floorNum + " from currentFloor " + currentFloor);
-        motor.move(floorNum);
-        logger.debug("Opening door of elevator:" + id);
+        motor.move(floorNum, passengerDestination);
+        //requested.removeFirst();
```
Comment (boriszugic) [inline src/Elevator.java:172]: Delete if not needed

File: src/ElevatorStateMachine.java [hunk 5]
```diff
@@ -9,22 +9,31 @@
...
+            context.setState(new Moving_down(state));
+        }
     }
 
     /**
      * Method displaying the current state.
      */
-    public void displayState(){
+    public void displayState() {
         System.out.println(this);
```
Comment (boriszugic) [inline src/ElevatorStateMachine.java:37]: Remove method

File: src/ElevatorStateMachine.java [hunk 6]
```diff
@@ -107,7 +133,64 @@ public void nextRequest(ElevatorStateMachine context){
...
+     *
+     * @param context Current context of the state machine.
+     */
+    @Override
+    public void requestReceived(ElevatorStateMachine context, String direction){}
+
+    /**
+     * Method displaying the current state.
+     */
+    public void displayState(){
```
Comment (boriszugic) [inline src/ElevatorStateMachine.java:156]: Remove method

File: src/ElevatorStateMachine.java [hunk 7]
```diff
@@ -116,22 +199,24 @@ public String toString(){
...
-    public void requestReceived(ElevatorStateMachine context){
-        context.setState(new Moving());
+    public void requestReceived(ElevatorStateMachine context, String direction){
+        context.setState(new UnloadingLoading(machine));
     }
 
     /**
      * Method displaying the current state.
      */
     public void displayState(){
```
Comment (boriszugic) [inline src/ElevatorStateMachine.java:218]: Remove method

File: src/Floor.java [hunk 8]
```diff
@@ -183,6 +183,7 @@ private void pressElevatorButton(int floorNum) {
      */
     private void board() {
         // get dest floor number from RequestData DS
+        System.out.println("request from" + floorNum);
```
Comment (boriszugic) [inline src/Floor.java:186]: Remove

File: src/Floor.java [hunk 9]
```diff
@@ -208,6 +209,7 @@ private void waitRequest(){
      * @param request  RequestData instance containing packet request information
      */
     public void printRequestInfo(RequestData request) {
+        System.out.println("FLOOR REQUEST" + request.toString());
```
Comment (boriszugic) [inline src/Floor.java:212]: Remove

File: src/Motor.java [hunk 10]
```diff
@@ -32,15 +34,29 @@ public Motor(Elevator e, ConfigurationReader config) {
...
         if (config.getNumFloors() < floorNum){
             return;
         }
         if (elevator.getCurrentFloor() == floorNum){
+            try {
+                TimeUnit.MILLISECONDS.sleep(config.getMovingTime());
+            }catch(InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+            System.out.println("Elevator is at " + elevator.getCurrentFloor());
```
Comment (boriszugic) [inline src/Motor.java:47]: Remove

File: src/Motor.java [hunk 11]
```diff
@@ -50,6 +66,8 @@ public void move(int floorNum) {
             elevator.setCurrentFloor(elevator.getCurrentFloor() +
                     (floorNum - elevator.getCurrentFloor() > 0 ? 1 : -1));
             elevator.getDisplay().display(String.valueOf(elevator.getCurrentFloor()));
+            System.out.println("Elevator moved to" + elevator.getCurrentFloor());
```
Comment (boriszugic) [inline src/Motor.java:69]: Remove

File: src/Scheduler.java [hunk 12]
```diff
@@ -181,9 +186,9 @@ private DatagramPacket waitRequest() {
      * @return The appropriate DatagramPacket to be sent
      */
     public DatagramPacket parseRequest(DatagramPacket packet){
+        //System.out.println("Packet length: "+packet.getLength());
```
Comment (boriszugic) [inline src/Scheduler.java:189]: Remove

File: src/Scheduler.java [hunk 13]
```diff
@@ -239,7 +244,7 @@ private DatagramPacket serveRequest(DatagramPacket packet) {
         int floorNum = data[0];
         // Create a packet to elevator port assigned to floor in the if statement above
         DatagramPacket returnPacket = createElevatorPacket(floorNum, floors.get((int)data[1]).getElevatorPort(), 0);
-        elevators.get(floors.get((int)data[1]).getElevatorPort()).setState(ElevatorStateEnum.IDLE);
+        //elevators.get(floors.get((int)data[1]).getElevatorPort()).setState(new IdleState());
```
Comment (boriszugic) [inline src/Scheduler.java:247]: Why is this commented out? Uncomment or remove

File: src/Scheduler.java [hunk 14]
```diff
@@ -276,16 +281,31 @@ private ElevatorStructure chooseElevator(Direction direction, int floorNum){
         Optional<ElevatorStructure> suitableElevator = elevs.stream()
                 .filter(elevator -> isElevatorSuitable(elevator, direction, floorNum))
                 .findFirst();
+        System.out.println("chooose Elevator is being ran here");
```
Comment (boriszugic) [inline src/Scheduler.java:284]: Remove

File: src/Scheduler.java [hunk 15]
```diff
@@ -276,16 +281,31 @@ private ElevatorStructure chooseElevator(Direction direction, int floorNum){
...
         // Set the new state of Scheduler's copy (Elevator will set its own)
         if(suitableElevator.isPresent()) {
+            ElevatorStateMachine stateMachine = suitableElevator.get().getState();
+            System.out.println(suitableElevator.get().getState().getState().toString());
```
Comment (boriszugic) [inline src/Scheduler.java:288]: Remove

File: src/Scheduler.java [hunk 16]
```diff
@@ -276,16 +281,31 @@ private ElevatorStructure chooseElevator(Direction direction, int floorNum){
...
+
             if (suitableElevator.get().getCurrFloor() - floorNum != 0) {
+                ElevatorState newState = suitableElevator.get().getCurrFloor() > floorNum ?
+                                                                                new Moving_up(stateMachine) :
+                                                                                new Moving_down(stateMachine);
+                stateMachine.setState(newState);
+                System.out.println("state after choose elevator ");
```
Comment (boriszugic) [inline src/Scheduler.java:295]: Remove

File: src/Scheduler.java [hunk 17]
```diff
@@ -276,16 +281,31 @@ private ElevatorStructure chooseElevator(Direction direction, int floorNum){
...
                         ElevatorStateEnum.MOVING_DOWN :
                         ElevatorStateEnum.MOVING_UP);
+                        */
+
                 logger.debug("Chose Elevator " + suitableElevator.get().getId());
             }
+
         }
         else{
+            System.out.println("grabbing the first in the elevators list");
```
Comment (boriszugic) [inline src/Scheduler.java:308]: Remove

File: src/Scheduler.java [hunk 18]
```diff
@@ -301,12 +321,15 @@ private ElevatorStructure chooseElevator(Direction direction, int floorNum){
      * @return True if elevator is suitable, false otherwise
      */
     private boolean isElevatorSuitable(ElevatorStructure elevator, Direction direction, int floorNum) {
-        switch (elevator.getState()) {
-            case IDLE:
+        System.out.println(elevator.getState().getState().toString());
```
Comment (boriszugic) [inline src/Scheduler.java:324]: Remove

File: src/Scheduler.java [hunk 19]
```diff
@@ -301,12 +321,15 @@ private ElevatorStructure chooseElevator(Direction direction, int floorNum){
...
+        switch(elevator.getState().getState().toString()) {
+            case "Idle":
                 return true; // An idle elevator is always suitable
-            case MOVING_UP:
+            case "Moving_up":
+                System.out.println("Moving UP Is recognized");
```
Comment (boriszugic) [inline src/Scheduler.java:329]: Remove

File: src/Scheduler.java [hunk 20]
```diff
@@ -301,12 +321,15 @@ private ElevatorStructure chooseElevator(Direction direction, int floorNum){
...
                 return direction == Direction.UP && elevator.getCurrFloor() <= floorNum;
-            case MOVING_DOWN:
+            case "Moving_down":
+                System.out.println("Moving DOWN is recognized");
```
Comment (boriszugic) [inline src/Scheduler.java:332]: Remove

File: src/Scheduler.java [hunk 21]
```diff
@@ -387,6 +410,9 @@ public void close(){
      * @param sender String name of the receiver
      */
     private void printPacketReceived(DatagramPacket packet, String sender) {
+        System.out.println("Packet received from " + sender);
+        System.out.println("From host port: " + packet.getPort());
+        System.out.println("Containing: "+ Arrays.toString(packet.getData()));
```
Comment (boriszugic) [inline src/Scheduler.java:415]: Remove

File: src/Scheduler.java [hunk 22]
```diff
@@ -399,6 +425,9 @@ private void printPacketReceived(DatagramPacket packet, String sender) {
      */
     private void printPacketRequest(DatagramPacket packet) {
         //Information prints
+        System.out.println("Sending packet:");
+        System.out.println("Destination host port: " + packet.getPort());
+        System.out.println("Containing: " + Arrays.toString(packet.getData())+"\n");
```
Comment (boriszugic) [inline src/Scheduler.java:430]: Remove

**Review by johnnynguyen3 — APPROVED** [review]

**Pull request merged**